src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -570,32 +602,126 @@ export class ResultsViewPageStore {
+                ids: _.map(entities, clinicalDataType === "SAMPLE" ? 'sampleId' : 'patientId')
+            };
+            return client.fetchAllClinicalDataInStudyUsingPOST({
+                studyId:study.studyId,
+                clinicalDataSingleStudyFilter: filter,
+                clinicalDataType: clinicalDataType
+            });
+        } else {
+            const filter: ClinicalDataMultiStudyFilter = {
+                attributeIds: attributeIds,
+                identifiers: entities.map((s: any) => clinicalDataType === "SAMPLE" ?
+                    ({entityId: s.sampleId, studyId: s.studyId}) : ({entityId: s.patientId, studyId: s.studyId}))
+            };
+            return client.fetchClinicalDataUsingPOST({
+                clinicalDataType: clinicalDataType,
+                clinicalDataMultiStudyFilter: filter
+            });
+        }
+    }
+
+    readonly survivalClinicalData = remoteData<ClinicalData[]>({
+        await: () => [
+            this.studies,
+            this.patients
+        ],
+        invoke: () => this.getClinicalData("PATIENT", this.patients.result, ["OS_STATUS", "OS_MONTHS", "DFS_STATUS", "DFS_MONTHS"])
+    }, []);
+
+    readonly survivalClinicalDataGroupByUniquePatientKey = remoteData<Dictionary<ClinicalData[]>>({
+        await: () => [
+            this.survivalClinicalData,
+        ],
+        invoke: async() => {
+            return _.groupBy(this.survivalClinicalData.result, 'uniquePatientKey');
+        }
+    });
+
+    private getPatientSurvivals(targetUniquePatientKeys: string[], statusAttributeId: string, monthsAttributeId: string,
+                                statusFilter: (s: string) => boolean): PatientSurvival[] {
+
+        let patientSurvivals: PatientSurvival[] = [];
+        if (targetUniquePatientKeys) {
+            targetUniquePatientKeys.forEach((uniquePatientKey: string) => {
+                const clinicalData: ClinicalData[] = this.survivalClinicalDataGroupByUniquePatientKey.result![uniquePatientKey];
+                if (clinicalData) {
+                    const statusClinicalData: ClinicalData[] = clinicalData.filter(c => c.clinicalAttributeId === statusAttributeId);
+                    const monthsClinicalData: ClinicalData[] = clinicalData.filter(c => c.clinicalAttributeId === monthsAttributeId);
+                    if (statusClinicalData[0] && monthsClinicalData[0] && statusClinicalData[0].value != 'NA' &&
+                        monthsClinicalData[0].value != 'NA') {
+                        const patient: Patient = this.patients.result.filter(p => p.uniquePatientKey === uniquePatientKey)[0];
+                        patientSurvivals.push({
+                            patientId: patient.patientId,
+                            studyId: patient.studyId,
+                            status: statusFilter(statusClinicalData[0].value),
+                            months: Number.parseFloat(monthsClinicalData[0].value)
```

Comment:
I don't think Number.parseFloat is not supported by IE.  It's possible that it's polyfilled by Babel, but I'm not sure.  You can probably just use the global parseFloat.